arm-network/2016-09-01/swagger/network.json

```diff
@@ -1663,7 +1663,7 @@
             "description": "The name of the network interface."
           },
           {
-            "name": "parameters",
+            "name": "properties",
```

Comment:
The changes suggested are the result of attempting to call the api with the request format defined at the api spec and receiving errors.
More specifically, calling the action 'CreateOrUpdatePublicIPAddresses' with the following parameters (which follow the spec):
```
{
	"publicIpAddressName": PUBLIC_IP_NAME,
	"resourceGroupName": RESOURCE_GROUP_NAME,
	"subscriptionId": SUBSCRIPTION_ID,
	"parameters": {
		"location": "westeurope",
		"publicIPAllocationMethod": "static"
	},
	"api-version": "2016-09-01"
}
```

Returns the following error:
```
{
	"code": "InvalidRequestContent",
	"message": "The request content was invalid and could not be deserialized: 'Could not find member 'parameters' on object of type 'ResourceDefinition'. Path 'parameters', line 1, position 14.'."
}
```

Applying the suggested change though, seems to fix this issue and the call does success.

Also, paying atention to the definition of "Resource", looks like the error message does indeed have its point:
https://github.com/Azure/azure-rest-api-specs/blob/master/arm-network/2016-09-01/swagger/network.json#L8172

Finally, and as @tbombach already mentioned, this change does match the documentation at:
https://msdn.microsoft.com/en-us/library/azure/mt163668.aspx